site/content/posts/2023-08-08-v0.25.0-impersonation-proxy-with-external-certs.md

```diff
@@ -0,0 +1,318 @@
+$ kubectl get secret local-user-authenticator-tls-serving-certificate \
+    --namespace local-user-authenticator \
+    --output jsonpath={.data.caCertificate} \
+    --kubeconfig kind-with-contour.kubeconfig.yaml \
+    | tee local-user-authenticator-ca.pem.b64
+```
+
+Install `cert-manager`:
+
+```shell
+$ kubectl apply \
+    --filename https://github.com/cert-manager/cert-manager/releases/download/v1.12.3/cert-manager.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Configure a `cert-manager` certificate for the impersonation proxy to serve TLS:
+
+```shell
+$ cat << EOF > self-signed-cert.yaml
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cert-manager
+---
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: selfsigned-issuer
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: my-selfsigned-ca
+  namespace: cert-manager
+spec:
+  isCA: true
+  commonName: my-selfsigned-ca
+  secretName: impersonation-proxy-self-signed-ca
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+  issuerRef:
+    name: selfsigned-issuer
+    kind: ClusterIssuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: my-ca-issuer
+spec:
+  ca:
+    secretName: root-secret
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: impersonation-serving-cert
+  namespace: pinniped-concierge
+spec:
+  secretName: impersonation-proxy-tls-serving-cert
+
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+    - Pinniped
+  isCA: false
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+  - server auth
+  dnsNames:
+  - impersonation-proxy-mtls.local
+  issuerRef:
+    name: my-ca-issuer
+    kind: ClusterIssuer
+    group: cert-manager.io
+
+EOF
+
+$ kubectl apply \
+    --filename self-signed-cert.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Download the root (self-signed) CA's certificate so that it can be advertised as the CA bundle for the Concierge Impersonation Proxy:
+
+```shell
+kubectl get secret root-secret \
+    --namespace cert-manager \
+    --output jsonpath="{.data.ca\.crt}" > impersonation-proxy-self-signed-ca.pem.b64
+    
+# Tip: Put the contents of impersonation-proxy-self-signed-ca.pem.b64 into your copy buffer for a later step!
+```
+
+Install Pinniped’s Concierge:
+
+```shell
+$ kubectl apply \
+    --filename https://get.pinniped.dev/v0.25.0/install-pinniped-concierge-crds.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+
+$ kubectl apply \
+    --filename https://get.pinniped.dev/v0.25.0/install-pinniped-concierge-resources.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Now update the `CredentialIssuer` to use the Impersonation Proxy (which is disabled on kind by default):
```

Comment:
Lets describe, perhaps at the top, how the impersonation proxy provides the ability to integrate with public clouds that might otherwise be impossible...

Reply:
I think linking to the docs on the impersonation proxy is sufficient.